contracts/src/TeleporterTokenDestination.sol

```diff
@@ -199,93 +199,13 @@ abstract contract TeleporterTokenDestination is
-                "TeleporterTokenDestination: invalid destination bridge address"
-            );
-            require(input.secondaryFee == 0, "TeleporterTokenDestination: non-zero secondary fee");
-            require(
-                input.multiHopFallback == address(0),
-                "TeleporterTokenDestination: non-zero multi-hop fallback"
-            );
-            message = BridgeMessage({
-                messageType: BridgeMessageType.SINGLE_HOP_SEND,
-                payload: abi.encode(SingleHopSendMessage({recipient: input.recipient, amount: amount}))
-            });
+            _processSend(input, amount);
         } else {
-            // Require a multi-hop fallback in case the message sent to the intermediate source
-            // chain fails to route the tokens to the final destination.
-            require(
-                input.multiHopFallback != address(0),
-                "TeleporterTokenDestination: zero multi-hop fallback"
-            );
-
-            // If the destination blockchain ID is this blockchian, the destination
-            // bridge address must be a differet contract. This is a multi-hop case to
-            // a different bridge contract on this chain.
-            if (input.destinationBlockchainID == blockchainID) {
-                require(
-                    input.destinationBridgeAddress != address(this),
-                    "TeleporterTokenDestination: invalid destination bridge address"
-                );
-            }
-            message = BridgeMessage({
-                messageType: BridgeMessageType.MULTI_HOP_SEND,
-                payload: abi.encode(
-                    MultiHopSendMessage({
-                        destinationBlockchainID: input.destinationBlockchainID,
-                        destinationBridgeAddress: input.destinationBridgeAddress,
-                        recipient: input.recipient,
-                        amount: amount,
-                        secondaryFee: input.secondaryFee,
-                        secondaryGasLimit: input.requiredGasLimit,
-                        multiHopFallback: input.multiHopFallback
-                    })
-                    )
-            });
-            messageRequiredGasLimit = MULTI_HOP_REQUIRED_GAS;
+            _processSendMultiHop(input, amount);
         }
-
-        bytes32 messageID = _sendTeleporterMessage(
-            TeleporterMessageInput({
-                destinationBlockchainID: sourceBlockchainID,
-                destinationAddress: tokenSourceAddress,
-                feeInfo: TeleporterFeeInfo({
-                    feeTokenAddress: input.primaryFeeTokenAddress,
-                    amount: primaryFee
-                }),
-                requiredGasLimit: messageRequiredGasLimit,
-                allowedRelayerAddresses: new address[](0),
-                message: abi.encode(message)
-            })
-        );
-
-        emit TokensSent(messageID, _msgSender(), input, amount);
```

Comment:
In both `_send` and `_sendAndCall`, the `_sendTeleporterMessage` call is repeated in the single hop and multihop helpers. There's an opportunity to reduce code duplication by returning the message params from the `_process*` helpers, and calling `_sendTeleporterMessage` in `_send`. There may be gas considerations with passing extra vars through the callstack though.